registry/auth.go

```diff
@@ -47,23 +48,23 @@ func loginV1(authConfig *types.AuthConfig, apiEndpoint APIEndpoint, userAgent st
 	defer resp.Body.Close()
 	body, err := ioutil.ReadAll(resp.Body)
 	if err != nil {
-		return "", "", systemError{err}
+		return "", "", errdefs.System(err)
 	}
 
 	switch resp.StatusCode {
 	case http.StatusOK:
 		return "Login Succeeded", "", nil
 	case http.StatusUnauthorized:
-		return "", "", unauthorizedError{errors.New("Wrong login/password, please try again")}
+		return "", "", errdefs.Forbidden(errors.New("Wrong login/password, please try again"))
```

Comment:
This should probably be `errdefs.Unauthorized()`

Reply:
I don't know why I chose this, but I think you are right.